support/catalogs/images.go

```diff
@@ -53,7 +53,7 @@ func GetCatalogImages(ctx context.Context, hcp hyperv1.HostedControlPlane, pullS
 	return getCatalogImagesWithCache(
 		imageLookupCacheKeyFn(&hcp, pullSecret, registryOverrides),
 		releaseVersionFn(ctx, &hcp, pullSecret, imageMetadataProvider),
-		imageExistsFn(ctx, pullSecret, imageMetadataProvider),
+		imageExistsFn(ctx, pullSecret, imageMetadataProvider, hcp.Spec.OLMCatalogPlacement != hyperv1.ManagementOLMCatalogPlacement),
```

Comment:
I think that you should pass `&hcp` to `imageExistsFn` like is done for the other functions.  Then within the function you can check the OLM catalog placement.